src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs

```diff
@@ -43,6 +45,50 @@ public static Task<RemoteHostClient> TryGetRemoteHostClientAsync(this Workspace
             return clientService?.TryGetRemoteHostClientAsync(cancellationToken);
         }
 
+        public static Task<RemoteHostClient.Session> TryCreateCodeAnalysisServiceSessionAsync(
+             this Solution solution, Option<bool> option, string experimentName, CancellationToken cancellationToken)
+             => TryCreateCodeAnalysisServiceSessionAsync(solution, option, experimentName, callbackTarget: null, cancellationToken: cancellationToken);
+
+        public static bool IsOutOfProcessEnabled(this Workspace workspace, Option<bool> option, string experimentName)
```

Comment:
Nit: Put below `TryCreateCodeAnalysisServiceSessionAsync`, and make a private non-extension method.